projects/plugins/boost/app/assets/src/js/features/page-cache/health/health.tsx

```diff
@@ -1,34 +1,61 @@
-import { Button, Notice } from '@automattic/jetpack-components';
+import { Notice } from '@automattic/jetpack-components';
 import getErrorData from './lib/get-error-data';
 import { __ } from '@wordpress/i18n';
-import { type PageCacheError } from '$lib/stores/page-cache';
-
+import { usePageCacheSetup, type PageCacheError } from '$lib/stores/page-cache';
+import { MutationNotice } from '$features/ui';
+import { useEffect, useState } from 'react';
+import { useSingleModuleState } from '$features/module/lib/stores';
 type HealthProps = {
-	error: PageCacheError;
-	setupCache: () => void;
+	error?: PageCacheError;
+	setup?: ReturnType< typeof usePageCacheSetup >;
+	setError: ( error: PageCacheError ) => void;
 };
 
-const Health = ( { error, setupCache }: HealthProps ) => {
+const Health = ( { setup, error, setError }: HealthProps ) => {
+	const [ pageCache, setPageCache ] = useSingleModuleState( 'page_cache' );
 	// Was there a problem trying to setup cache?
 	const errorData = getErrorData( error );
-	if ( errorData ) {
-		return (
-			<Notice
-				level="warning"
-				hideCloseButton={ true }
-				title={ errorData.title }
-				actions={ [
-					<Button size="small" weight="regular" onClick={ setupCache } key="try-again">
-						{ __( 'Try again', 'jetpack-boost' ) }
-					</Button>,
-				] }
-			>
-				<p>{ errorData.message }</p>
-			</Notice>
-		);
-	}
 
-	return null;
+	const [ canReset, setCanReset ] = useState( false );
+	useEffect( () => {
+		if ( setup?.isError && error && ! error.dismissed ) {
+			setCanReset( true );
+		}
+	}, [ setup?.isError, error, setCanReset ] );
+	useEffect( () => {
+		if ( canReset ) {
+			setCanReset( false );
+			setPageCache( false );
+		}
+	}, [ canReset, setCanReset, setPageCache ] );
+	return (
+		<>
+			{ setup && (
+				<MutationNotice
+					mutationId="page-cache-setup"
+					isPending={
+						setup.isPending || ( !! pageCache?.active && ( ! error || !! error.dismissed ) )
```

Comment:
I think this causes the `Setting up cache...` to show up when opening the page and caching is enabled (and working).

![image](https://github.com/Automattic/jetpack/assets/11799079/f291916d-3067-42d6-bf54-6b6da6b8886e)

To reproduce, enable caching and wait for setup to complete. Then refresh the page and you should see the same notice.

Reply:
With 112d204 this no longer happens.